plugins/github-actions/src/components/WorkflowRunsCard/WorkflowRunsCard.tsx

```diff
@@ -0,0 +1,417 @@
+                    >
+                      <MarkdownContent
+                        content={`Commit ID : ${run.source.commit.hash!}`}
+                      />
+                    </Typography>
+                  </Tooltip>
+
+                  {run.source.branchName && (
+                    <MarkdownContent
+                      content={`Branch : ${run.source.branchName}`}
+                    />
+                  )}
+                  <Chip
+                    key={run.id}
+                    size="small"
+                    label={`Workflow ID : ${run.id}`}
+                  />
+                  <Chip
+                    size="small"
+                    label={
+                      <Box display="flex" alignItems="center">
+                        <WorkflowRunStatus
+                          status={run.status}
+                          conclusion={run.conclusion}
+                        />
+                      </Box>
+                    }
+                  />
+                  <div className={classes.bottomline}>
+                    {run.githubUrl && (
+                      <Link to={run.githubUrl}>
+                        Workflow runs on GitHub{' '}
+                        <ExternalLinkIcon
+                          className={classes.externalLinkIcon}
+                        />
+                      </Link>
+                    )}
+                    <ButtonGroup>
+                      <Tooltip title="Rerun workflow">
+                        <IconButton
+                          onClick={run.onReRunClick}
+                          style={{ fontSize: '12px' }}
+                        >
+                          <RetryIcon />
+                        </IconButton>
+                      </Tooltip>
+                    </ButtonGroup>
+                  </div>
+                </Box>
+              </Box>
+            </Box>
+          </Grid>
+        ))
+      ) : (
+        <Box p={2}>
+          {loading ? <CircularProgress /> : 'No matching runs found.'}
+        </Box>
+      )}
+      <div className={classes.pagination}>
+        <TablePagination
+          component="div"
+          count={total}
+          page={page}
+          rowsPerPage={pageSize}
+          onPageChange={(_, newPage) => onChangePage(newPage)}
+          onRowsPerPageChange={event =>
+            onChangePageSize(parseInt(event.target.value, 6))
+          }
+          labelRowsPerPage="Workflows per page"
+          rowsPerPageOptions={[6, 12, 18, { label: 'All', value: -1 }]}
+        />
+      </div>
+    </Grid>
+  );
+};
+
+type WorkflowRunsCardProps = {
+  entity: Entity;
+};
+
+const WorkflowRunsCardSearch = ({
+  searchTerm,
+  handleSearch,
+  retry,
+}: {
+  searchTerm: string;
+  handleSearch: (event: ChangeEvent<HTMLInputElement>) => void;
+  retry: (() => void) | (() => void) | (() => void) | (() => void);
+}) => {
+  return (
+    <>
+      <Box flexGrow={1} />
+      <TextField
+        id="search"
```

Comment:
It's better to avoid IDs if possible https://medium.com/@clairecodes/reasons-not-to-use-ids-in-css-a42204fb0d97#:~:text=Classes%20can%20be%20reused,of%20the%20advantages%20of%20CSS.

Reply:
@acierto, I'm using the IDs part of logic to get the search control and drop menu control, so it's needed for now, can we make this change part of next phase improvement?

Reply:
+1 to avoiding `id`. I'm not seeing where you end up using it btw? Typically you'd be able to use a ref instead.

Reply:
Fixed, my bad Id is not used anywhere.